esmvaltool/cmorizers/data/formatters/datasets/esdc.py

```diff
@@ -0,0 +1,117 @@
+"""ESMValTool CMORizer for Earth System Data Cube data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+    http://data.rsc4earth.de/EarthSystemDataCube/
+
+Last access
+    20220927
+
+Download and processing instructions
+    It is not necessary to download the data, as the cmorizer script can access
+    it directly from the cloud if it is not available locally.
+
+    To download a dataset, the dataset folder can be explored on the source
+    website, and downloaded using wget:
+        ```wget -m -nH http://url/to/dataset```
+"""
+import logging
+from pathlib import Path
+
+import xarray as xr
+from esmvalcore.preprocessor import monthly_statistics
+
+from esmvaltool.cmorizers.data import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def fix_cube(short_name, var, cube, cfg):
+    """General fixes for all cubes."""
+    cmor_info = cfg['cmor_table'].get_variable(var['mip'], short_name)
+    utils.fix_coords(cube)
+    cube.convert_units(cmor_info.units)
+    if 'height2m' in cmor_info.dimensions:
+        utils.add_height2m(cube)
+    cube = monthly_statistics(cube, operator="mean")  # Fix frequency
+    # Fix metadata
+    attrs = cfg['attributes']
+    attrs['mip'] = var['mip']
+    utils.fix_var_metadata(cube, cmor_info)
+    utils.set_global_atts(cube, attrs)
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize the dataset."""
+    # 1. find the input data
+    logger.debug("in_dir: '%s'", in_dir)
+    logger.debug("cfg: '%s'", cfg)
+    logger.debug("cfg_user: '%s'", cfg_user)
+
+    if start_date:
+        logger.warning('start_date set to "%s", but will be ignored',
+                       start_date)
+    if end_date:
+        logger.warning('end_date set to "%s", but will be ignored', end_date)
+
+    attributes = cfg['attributes']
+    variables = cfg['variables']
+
+    version = attributes['version']
+
+    filename_pattern = cfg['filename'].format(grid=attributes['grid'],
+                                              chunking=attributes['chunking'],
+                                              version=version)
+
+    def cmorize_cube(path):
```

Comment:
I have formatted that function. It feels a bit awkward to loop over the variables before opening the zarr cube, but it seems it doesn't have a severe performance penalty, and it could be parallelized following the example of https://github.com/ESMValGroup/ESMValTool/blob/5aef74c22e825526e910a5b49056c9d65efba7e8/esmvaltool/cmorizers/data/formatters/datasets/era_interim.py